File: backend/app/api/v1/expense.py
"""Expense API - 支出记录接口"""
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.database import get_db
from app.core.deps import get_current_user, rate_limit_general
from app.core.exceptions import success_response
from app.models.user import User
from app.schemas.expense_record import ExpenseListCreate, ExpenseRecordResponse
from app.services.expense_service import create_expense_records, get_expenses_by_salary

router = APIRouter(prefix="/salary", tags=["expenses"])

@router.post("/{recordId}/expenses", dependencies=[Depends(rate_limit_general)])
async def create_expenses(recordId: str, body: ExpenseListCreate, current_user: User = Depends(get_current_user), db: AsyncSession = Depends(get_db)):
    records = await create_expense_records(db, current_user.id, recordId, [exp.model_dump() for exp in body.expenses])
    response = [ExpenseRecordResponse(id=r.id, salaryRecordId=r.salary_record_id, expenseDate=r.expense_date, category=r.category, subcategory=r.subcategory, amount=float(r.amount), note=r.note, createdAt=r.created_at) for r in records]
    return success_response(data={"records": response, "total": len(response)}, message="支出记录创建成功")

@router.get("/{recordId}/expenses")
async def get_expenses(recordId: str, current_user: User = Depends(get_current_user), db: AsyncSession = Depends(get_db)):
    records = await get_expenses_by_salary(db, recordId)
    response = [ExpenseRecordResponse(id=r.id, salaryRecordId=r.salary_record_id, expenseDate=r.expense_date, category=r.category, subcategory=r.subcategory, amount=float(r.amount), note=r.note, createdAt=r.created_at) for r in records]
    return success_response(data={"records": response, "total": len(response)})

File: miniapp/src/api/expense.ts
/**
 * 支出记录 - Sprint 4.2
 * 与 backend /api/v1/salary/{recordId}/expenses 一致
 */
import request from '@/utils/request'

const PREFIX = '/api/v1/salary'

/** 支出记录创建项 */
export interface ExpenseRecordCreate {
  expenseDate: string
  category: string
  subcategory?: string
  amount: number
  note?: string
}

/** 支出记录响应 */
export interface ExpenseRecordResponse {
  id: string
  salaryRecordId: string
  expenseDate: string
  category: string
  subcategory?: string
  amount: number
  note?: string
  createdAt: string
}

/** 批量创建请求 */
export interface ExpenseListCreate {
  expenses: ExpenseRecordCreate[]
}

/** 创建支出记录响应 */
export interface ExpenseListResponse {
  records: ExpenseRecordResponse[]
  total: number
}

/** 创建支出记录 */
export function createExpenses(recordId: string, data: ExpenseListCreate) {
  return request<ExpenseListResponse>({
    url: `${PREFIX}/${recordId}/expenses`,
    method: 'POST',
    data,
  })
}

/** 获取支出记录 */
export function getExpenses(recordId: string) {
  return request<ExpenseListResponse>({
    url: `${PREFIX}/${recordId}/expenses`,
    method: 'GET',
  })
}
